tests/views/test.sign-in.jsx

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import TestUtils from 'react/lib/ReactTestUtils';
+
+import { initialUserState } from 'reducers/user';
+import SignIn from 'views/shared/sign-in';
+
+
+describe('SignIn', function() {
+
+  var fakeAccessToken = 'some-oauth-access-token';
+  var fakeTokenSignIn;
+  var fakeUser;
+  var fakeUserSignIn;
+
+  beforeEach(function() {
+    fakeTokenSignIn = sinon.spy();
+    fakeUser = Object.assign({}, initialUserState);
+    fakeUserSignIn = sinon.spy();
+  });
+
+  function mountView({accessToken=fakeAccessToken, user=fakeUser} = {}) {
+    return TestUtils.renderIntoDocument(
+      <SignIn
+        accessToken={accessToken}
+        tokenSignIn={fakeTokenSignIn}
+        user={user}
+        userSignIn={fakeUserSignIn}
+      />
+    );
+  }
+
+  it('should sign in on mount', function() {
+    mountView();
+    assert.equal(fakeTokenSignIn.firstCall.args[0], fakeAccessToken);
+  });
+
+  it('should do a token sign-in with access token', function() {
+    var token = 'some-fxa-token';
+    mountView({accessToken: token});
+    assert.equal(fakeTokenSignIn.firstCall.args[0], token);
+  });
+
+  it('should do a user sign-in without access token', function() {
+    mountView({accessToken: null});
```

Comment:
Yeah that makes sense because that's the same thing as a key not being present at all.

Seeing the exported transpilation is quite helpful for visualisation:

https://babeljs.io/repl/#?experimental=true&evaluate=true&loose=false&spec=false&code=function%20mountView%28{accessToken%3DfakeAccessToken%2C%20user%3DfakeUser}%20%3D%20{}%29%20{%0A%20%20%0A}